core/src/wallet/tezos/api_impl/TezosLikeTransactionApi.cpp

```diff
@@ -364,6 +367,197 @@ namespace ledger {
             }
             return writer.toByteArray();
         }
+
+        std::vector<uint8_t> TezosLikeTransactionApi::serializeForDryRun(const std::vector<uint8_t>& chainId) {
+            BytesWriter writer;
+            writer.writeByteArray(serialize());
+            writer.writeByteArray(chainId);
+            return writer.toByteArray();
+        }
+
+        std::string TezosLikeTransactionApi::serializeJsonForDryRun(const std::string &chainID)
+        {
+            using namespace rapidjson;
+
+            Value vString(kStringType);
+            Document tx;
+            tx.SetObject();
+            Document::AllocatorType &allocator = tx.GetAllocator();
+
+            // Chain Id
+            vString.SetString(chainID.c_str(), static_cast<SizeType>(chainID.length()), allocator);
+            tx.AddMember("chain_id", vString, allocator);
+
+            // Operation
+            Value opObject(kObjectType);
+            {
+                // Branch
+                const auto hash = _block->getHash();
+                vString.SetString(hash.c_str(), static_cast<SizeType>(hash.length()), allocator);
+                opObject.AddMember("branch", vString, allocator);
+
+                // Fake sign
```

Comment:
hardcoded value ?

Reply:
Fake sign is hardcoded because the signature must be in a format that tezos node can parse, even if it’s never verified